openml/datasets/dataset.py

```diff
@@ -75,7 +76,7 @@ def __init__(self, id=None, name=None, version=None, description=None,
                 logger.debug("Data pickle file already exists.")
             else:
                 try:
-                    data = self._get_arff()
```

Comment:
why is ``format`` an attribute? Can we find that out from the file? Shouldn't the file reader handle this?